cli/internal/k8s/gitops.go

```diff
@@ -0,0 +1,164 @@
+package k8s
+
+// Mostly taken from https://github.com/argoproj/gitops-engine/blob/master/agent/main.go
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	adapter "github.com/bombsimon/logrusr"
+	"github.com/go-logr/logr"
+
+	"github.com/argoproj/gitops-engine/pkg/cache"
+	"github.com/argoproj/gitops-engine/pkg/engine"
+	"github.com/argoproj/gitops-engine/pkg/sync"
+	"github.com/argoproj/gitops-engine/pkg/utils/kube"
+	"github.com/defenseunicorns/zarf/cli/internal/utils"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	annotationGCMark = "gitops-agent.argoproj.io/gc-mark"
+)
+
+type resourceInfo struct {
+	gcMark string
+}
+
+type settings struct {
+	path string
+}
+
+func (syncSettings *settings) getGCMark(key kube.ResourceKey) string {
+	h := sha256.New()
+	_, _ = h.Write([]byte(syncSettings.path))
+	_, _ = h.Write([]byte(strings.Join([]string{key.Group, key.Kind, key.Name}, "/")))
+	return "sha256." + base64.RawURLEncoding.EncodeToString(h.Sum(nil))
+}
+
+func (syncSettings *settings) parseManifests() ([]*unstructured.Unstructured, error) {
+	var res []*unstructured.Unstructured
+
+	manifests := utils.RecursiveFileList(syncSettings.path)
+
+	for _, manifest := range manifests {
+		if ext := strings.ToLower(filepath.Ext(manifest)); ext == ".yml" || ext == ".yaml" {
+			// Load the file contents
+			data, err := ioutil.ReadFile(manifest)
+			if err != nil {
+				logrus.Fatal(err)
```

Comment:
I think we're being a little inconsistent on when we're adding the err object to our Fatal logs and when we're just logging a simple error message. 

Now that we can set log-levels I would vote towards logging simpler fatal messages (but adding file paths if a local file lead to an error) and throwing the `err` in a debug log.

Reply:
yeah that's a good catch, this is still a huge refactor and needs cleaning up

Reply:
feel free to just make those kind of changes and push to this branch